main.go

```diff
@@ -1547,6 +1560,159 @@ func (c MainController) ListAllChapters(w http.ResponseWriter, r *http.Request)
 	writeJSON(w, chapters)
 }
 
+// testing this
+func (c MainController) discordBotAuthMiddleware(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		err := r.ParseForm()
+		if err != nil {
+			http.Error(w, http.StatusText(400), 400)
+			return
+		}
+
+		// check that discord auth matches (shared secret w/ discord bot)
+		discordAuth := r.PostFormValue("auth")
+		if discordAuth != config.DiscordSecret {
+			http.Error(w, http.StatusText(401), 401)
+			return
+		}
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+func (c MainController) DiscordStatusHandler(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+	if err != nil {
+		writeJSON(w, map[string]interface{}{
+			"status": "error parsing form",
+		})
+		return
+	}
+
+	discordUserID, err := strconv.Atoi(r.PostFormValue("id"))
+
+	// see if a record exists for this user id in the discord_users table (pending, confirmed, or not found)
+	status, err := model.GetDiscordUserStatus(c.db, discordUserID)
+	if err != nil {
+		panic(err.Error())
+	}
+	log.Println("Discord user status:", status)
+
+	// return the status found from database
+	writeJSON(w, map[string]interface{}{
+		"status": status,
+	})
+}
+
+func (c MainController) DiscordGenerateHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, http.StatusText(400), 400)
+		return
+	}
+
+	var user model.DiscordUser
+	userID, err := strconv.Atoi(r.PostFormValue("id"))
+	if err != nil {
+		http.Error(w, http.StatusText(400), 400)
+		return
+	}
+	user.ID = userID
+	user.Email = r.PostFormValue("email")
+	user.Token = generateToken()
+
+	// confirm that this email exists in ADB. if not, return error.
+	activists, err := model.GetActivistsByEmail(c.db, user.Email)
+	if len(activists) < 1 {
+		writeJSON(w, map[string]interface{}{
+			"status": "invalid email",
+		})
+		return
+	}
+
+	// check that user isn't already confirmed
+	status, err := model.GetDiscordUserStatus(c.db, user.ID)
+	if err != nil {
+		panic(err.Error())
+	}
+	if status == model.Confirmed {
+		writeJSON(w, map[string]interface{}{
+			"status": "already confirmed",
+		})
+		return
+	}
+
+	// INSERT/REPLACE into database (only allows one record per discord ID)
+	err = model.InsertOrUpdateDiscordUser(c.db, user)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// trigger email to be sent w/ verification link
+	if config.DiscordFromEmail != "" && config.AWSAccessKey != "" && config.AWSSecretKey != "" && config.AWSSESEndpoint != "" {
+		subjectText := "Please verify your email"
+		// TODO: make nicer looking email
+		confirmLink := config.UrlPath + "/discord/confirm/" + strconv.Itoa(user.ID) + "/" + user.Token
+		bodyText := "Hello, Please click this link to verify your email address: " + confirmLink + " Cheers, The DxE Discord Bot"
```

Comment:
It would probably be good to add some newlines before and after the link so it's easier to click.

Reply:
This is just for testing everything right now. I'll make the email look nicer later.